src/main/java/ch/njol/skript/Skript.java

```diff
@@ -972,16 +977,71 @@ public static Metrics getMetrics() {
 	public static void closeOnDisable(final Closeable closeable) {
 		closeOnDisable.add(closeable);
 	}
-	
+
+	@SuppressWarnings("unused")
+	@EventHandler
+	public void onPluginDisable(PluginDisableEvent event) {
+		Plugin plugin = event.getPlugin();
+		PluginDescriptionFile descriptionFile = plugin.getDescription();
+		if (descriptionFile.getDepend().contains("Skript") || descriptionFile.getSoftDepend().contains("Skript")) {
+			// An addon being disabled, check if server is being stopped
+			if (!isServerRunning()) {
+				beforeDisable();
+			}
+		}
+	}
+
+	private static final Method IS_RUNNING;
+	private static final Object MC_SERVER;
+
+	static {
+		Server server = Bukkit.getServer();
+		Class<?> clazz = server.getClass();
+
+		Method serverMethod;
+		try {
+			serverMethod = clazz.getMethod("getServer");
+		} catch (NoSuchMethodException e) {
+			throw new RuntimeException(e);
+		}
+
+		try {
+			MC_SERVER = serverMethod.invoke(server);
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new RuntimeException(e);
+		}
+
+		try {
+			IS_RUNNING = MC_SERVER.getClass().getMethod("isRunning");
```

Comment:
Paper actually has a public `Server#isStopping` method. can we try to use that when on paper? https://papermc.io/javadocs/paper/1.16/org/bukkit/Server.html#isStopping()